File: Python3/string_compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        count = 1
                
        for i in range(len(chars)-1, -1, -1):
            if i and chars[i] == chars[i-1]: 
                count += 1
                chars.pop(i)
            else:
                if count > 1:
                    for num in str(count)[::-1]:
                        chars.insert(i+1, num)
                count = 1
        
        return len(chars)

File: C++/string_compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int count = 1;
                
        for(int i = chars.size()-1; i > -1; i--) {
            if(i && chars[i] == chars[i-1]) { 
                count ++;
                chars.erase(chars.begin()+i);
            } else {
                if(count > 9) {
                    chars.insert(chars.begin()+i+1, (count%10)+'0');
                    chars.insert(chars.begin()+i+1, (count/10)+'0');
                }
                else if(count > 1 && count <= 9) chars.insert(chars.begin()+i+1, count+'0');
                count = 1;
            }
        }
        return chars.size();
    }
};
